calcgoldplus.rb

```diff
@@ -0,0 +1,92 @@
+# this is the gold calculator with the float identifier element of platinum.
+# get the operation
+
+puts "Welcome to your calculator! What kind of operation would you like to do?"
+operation = gets.chomp.downcase
+
+# operation logic, accounting for multiple entry types
+
+if operation == "add" || operation == "+" || operation == "addition"
+  puts "Great, addition!"
+  operation = "add"
+elsif operation == "subtract" || operation == "-" || operation == "subtraction"
+  puts "Great, subtraction!"
+  operation = "sub"
+elsif operation == "multiply" || operation == "*" || operation == "multiplication"
+  puts "Great, multiplication!"
+  operation = "multi"
+elsif operation == "divide" || operation == "/" || operation == "division"
+  puts "Great, division!"
+  operation = "div"
+elsif operation == "exponent" || operation == "^" || operation == "power"
+  puts "Great, exponents!"
+  operation = "exp"
+elsif operation == "modulus" || operation == "%" || operation == "mod"
+  puts "Great, moduli!"
+  operation = "mod"
+else
+  puts "I'm sorry, I don't think you understand what I'm designed for,"
+  operation = "wrong"
+end
+
+# get numbers, account for string or nil input, while loop to re-ask for numeric input
+
+if operation != "wrong"
+
+  puts "What would you like your first number to be?"
+    num1 = gets.chomp
+
+  while num1 != "0" && num1.to_i == 0
```

Comment:
way to go to account for the fact that when you call to_i on a string that doesn't have numbers, it returns 0!  I might add a comment here just so that the person reading understands what you're doing here.  What happens if the user inputs 0.1?